test/e2e/suites/base/rescheduling_test.go

```diff
@@ -105,35 +121,22 @@ var _ = ginkgo.Describe("[cluster unjoined] reschedule testing", func() {
 			})
 		})
 
-		ginkgo.It("deployment reschedule testing", func() {
-			ginkgo.By(fmt.Sprintf("Joining cluster: %s", newClusterName), func() {
-				opts := join.CommandJoinOption{
-					DryRun:            false,
-					ClusterNamespace:  "karmada-cluster",
-					ClusterName:       newClusterName,
-					ClusterContext:    clusterContext,
-					ClusterKubeConfig: kubeConfigPath,
-				}
-				err := opts.Run(f)
-				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-
-				// wait for the current cluster status changing to true
-				framework.WaitClusterFitWith(controlPlaneClient, newClusterName, func(cluster *clusterv1alpha1.Cluster) bool {
-					return meta.IsStatusConditionPresentAndEqual(cluster.Status.Conditions, clusterv1alpha1.ClusterConditionReady, metav1.ConditionTrue)
-				})
-			})
-
+		ginkgo.It("propagate Deployment and waiting for reschedule", func() {
 			framework.CreatePropagationPolicy(karmadaClient, policy)
 			framework.CreateDeployment(kubeClient, deployment)
 			ginkgo.DeferCleanup(func() {
 				framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
 				framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
 			})
 
-			targetClusterNames := framework.ExtractTargetClustersFrom(controlPlaneClient, deployment)
+			ginkgo.By("check whether the deployment is rescheduled to a new cluster", func() {
```

Comment:
```suggestion
			ginkgo.By("check whether the deployment is scheduled to a new cluster", func() {
```

Creating clusters happens before deploying, so it should not involve rescheduling. Please correct me if I'm wrong.

Reply:
Thanks for reminding me, what I want to test is rescheduling when the cluster joins. I updated the logic, please help review again.